guide/samples/tests/mocking_lros.rs

```diff
@@ -0,0 +1,246 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Examples showing how to mock a client in tests.
+
+// ANCHOR: all
+#[cfg(test)]
+mod test {
+    use google_cloud_gax as gax;
+    use google_cloud_longrunning as longrunning;
+    use google_cloud_speech_v2 as speech;
+    use google_cloud_wkt as wkt;
+    type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
+
+    // ANCHOR: mockall-macro
+    mockall::mock! {
+        #[derive(Debug)]
+        Speech {}
+        impl speech::stub::Speech for Speech {
+            async fn batch_recognize(&self, req: speech::model::BatchRecognizeRequest, _options: gax::options::RequestOptions) -> gax::Result<longrunning::model::Operation>;
+            async fn get_operation(&self, req: longrunning::model::GetOperationRequest, _options: gax::options::RequestOptions) -> gax::Result<longrunning::model::Operation>;
+        }
+    }
+    // ANCHOR_END: mockall-macro
+
+    // ANCHOR: expected-response
+    fn expected_duration() -> Option<wkt::Duration> {
+        Some(wkt::Duration::clamp(100, 0))
+    }
+
+    fn expected_response() -> speech::model::BatchRecognizeResponse {
+        speech::model::BatchRecognizeResponse::new().set_total_billed_duration(expected_duration())
+    }
+    // ANCHOR_END: expected-response
+
+    // ANCHOR: auto-fn
+    // An example application function that automatically polls.
+    //
+    // It starts an LRO, awaits the result, and processes it.
+    async fn my_automatic_poller(
+        client: &speech::client::Speech,
+    ) -> gax::Result<Option<wkt::Duration>> {
+        use speech::Poller;
+        client
+            .batch_recognize("invalid-test-recognizer")
+            .poller()
+            .until_done()
+            .await
+            .map(|r| r.total_billed_duration)
+    }
+    // ANCHOR_END: auto-fn
+
+    // ANCHOR: finished-op
+    fn make_finished_operation(
+        response: &speech::model::BatchRecognizeResponse,
+    ) -> Result<longrunning::model::Operation> {
+        let any = wkt::Any::try_from(response)?;
```

Comment:
You could return `gax::Result` from here and do the `map_err(gax::Error::serde)` and simplify some code below

Reply:
Good catch, fixed, thanks.